src/main/scala/org/ergoplatform/nodeView/state/SnapshotsInfo.scala

```diff
@@ -0,0 +1,36 @@
+package org.ergoplatform.nodeView.state
+
+import org.ergoplatform.ErgoLikeContext.Height
+import org.ergoplatform.nodeView.state.UtxoState.ManifestId
+
+/**
+  * Container for UTXO set snapshots the node holds
+  * @param availableManifests - available UTXO set snapshot manifests and corresponding heights
+  */
+case class SnapshotsInfo(availableManifests: Map[Height, ManifestId]) {
```

Comment:
I suggest to avoid using arrays (which is runtime representation of ManifestId) directly in case classes. Arrays don't have equality, as result equality of case class becomes broken.
As an alternative it is better to use Seq[Byte] (or TaggedType over it) or better yet implement ManifestId as a wrapper over Digest32.

Reply:
It is not used as case class field here though

Reply:
It is used in the Map, and hence comparing two maps is broken.

Reply:
reworked into class , as equality is not needed